File: jambeth/jambeth-security/src/main/java/de/osthus/ambeth/privilege/model/impl/AllowAllPrivilege.java
package de.osthus.ambeth.privilege.model.impl;

import de.osthus.ambeth.privilege.model.IPrivilege;
import de.osthus.ambeth.privilege.model.IPropertyPrivilege;

public final class AllowAllPrivilege extends AbstractPrivilege
{
	public static final IPrivilege INSTANCE = new AllowAllPrivilege();

	private static final IPropertyPrivilege allowAllPropertyPrivilege = PropertyPrivilegeImpl.create(true, true, true, true);

	private AllowAllPrivilege()
	{
		super(true, true, true, true, true, null, null);
	}

	@Override
	public IPropertyPrivilege getDefaultPropertyPrivilegeIfValid()
	{
		return allowAllPropertyPrivilege;
	}

	@Override
	public IPropertyPrivilege getPrimitivePropertyPrivilege(int primitiveIndex)
	{
		return allowAllPropertyPrivilege;
	}

	@Override
	public IPropertyPrivilege getRelationPropertyPrivilege(int relationIndex)
	{
		return allowAllPropertyPrivilege;
	}

	@Override
	public boolean isCreateAllowed()
	{
		return true;
	}

	@Override
	public boolean isReadAllowed()
	{
		return true;
	}

	@Override
	public boolean isUpdateAllowed()
	{
		return true;
	}

	@Override
	public boolean isDeleteAllowed()
	{
		return true;
	}

	@Override
	public boolean isExecuteAllowed()
	{
		return true;
	}
}

File: ambeth/Ambeth.Security/ambeth/privilege/model/impl/AllowAllPrivilege.cs
using De.Osthus.Ambeth.Util;
using System;
using System.Text;

namespace De.Osthus.Ambeth.Privilege.Model.Impl
{
    public class AllowAllPrivilege : AbstractPrivilege
    {
        public static IPrivilege INSTANCE = new AllowAllPrivilege();

        private static IPropertyPrivilege allowAllPropertyPrivilege = PropertyPrivilegeImpl.Create(true, true, true, true);

        private AllowAllPrivilege() : base(true, true, true, true, true, null, null)
        {
            // intended blank
        }

        public override IPropertyPrivilege GetDefaultPropertyPrivilegeIfValid()
        {
            return allowAllPropertyPrivilege;
        }

        public override IPropertyPrivilege GetPrimitivePropertyPrivilege(int primitiveIndex)
        {
            return allowAllPropertyPrivilege;
        }

        public override IPropertyPrivilege GetRelationPropertyPrivilege(int relationIndex)
        {
            return allowAllPropertyPrivilege;
        }

        public override bool CreateAllowed
        {
            get { return true; }
        }

        public override bool ReadAllowed
        {
            get { return true; }
        }

        public override bool UpdateAllowed
        {
            get { return true; }
        }

        public override bool DeleteAllowed
        {
            get { return true; }
        }

        public override bool ExecuteAllowed
        {
            get { return true; }
        }
    }
}